.github/workflows/welcome.yml

```diff
@@ -7,7 +7,7 @@ on:
 jobs:
   welcome:
     runs-on: ubuntu-latest
-
+    if: ${{ github.actor != 'renovate[bot]' && github.actor != 'camperbot' }}
```

Comment:
The `github.actor` is the person who triggers the event, I think.

Which means if I merge the PR, I'm the actor.

Reply:
Well [this happened](https://github.com/freeCodeCamp/freeCodeCamp/runs/7026880841) after this PR was merged: https://github.com/freeCodeCamp/freeCodeCamp/pull/46626 - So I think the actor is the PR OP

Reply:
And we have been using this check on most of our workflows like codesee, codeql etc.

Reply:
Right, because our other work flows run on the `pull_request` event with the default types:

> For example, if no activity types are specified, the workflow runs when a pull request is opened or reopened or when the head branch of the pull request is updated.

So those actions run when the PR opens, and the actor would be the author.

In this action, because it runs on merge, I have to fetch the PR data to get the author:

```
const pull = await github.rest.pulls.get({
  owner: context.repo.owner,
  repo: context.repo.repo,
  pull_number: context.issue.number
});
const creator = pull.data.user.login;
```

What I could do is update the script in the action to check that `creator` value against our excluded bots - that should resolve the error.

Reply:
Ah - I see what you mean. Please go ahead. I will close this.